executor/join.go

```diff
@@ -269,6 +281,12 @@ func (e *HashJoinExec) fetchInnerRows(ctx context.Context, chkCh chan<- *chunk.C
 				return
 			}
 			if chk.NumRows() == 0 {
+				// Calculate the bit number needed when using radix partition.
+				if e.UseRadixPartition {
+					innerResultSize := float64(e.innerResult.GetMemTracker().BytesConsumed())
+					l2CacheSize := float64(e.ctx.GetSessionVars().L2CacheSize)
+					e.radixBits = int(math.Log2(innerResultSize / l2CacheSize))
```

Comment:
Should we round up the final result? What if `innerResultSize / l2CacheSize` is less than one? Maybe you need `bits.Len32()`.

Reply:
`radixBits == 0` means the inner table can fit into L2 cache so we skip partition phase?

Reply:
Yes @eurekaka @lamxTyler